cdap-app-templates/cdap-etl/cdap-data-streams-base/src/main/java/io/cdap/cdap/datastreams/SparkStreamingPipelineRunner.java

```diff
@@ -49,42 +59,70 @@
 import io.cdap.cdap.etl.spark.streaming.function.CountingTransformFunction;
 import io.cdap.cdap.etl.spark.streaming.function.DynamicJoinMerge;
 import io.cdap.cdap.etl.spark.streaming.function.DynamicJoinOn;
+import io.cdap.cdap.etl.spark.streaming.function.StreamingBatchSinkFunction;
+import io.cdap.cdap.etl.spark.streaming.function.StreamingMultiSinkFunction;
+import io.cdap.cdap.etl.spark.streaming.function.StreamingSparkSinkFunction;
 import io.cdap.cdap.etl.spark.streaming.function.WrapOutputTransformFunction;
 import io.cdap.cdap.etl.spark.streaming.function.preview.LimitingFunction;
 import io.cdap.cdap.etl.validation.LoggingFailureCollector;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.sql.SQLContext;
+import org.apache.spark.streaming.Time;
 import org.apache.spark.streaming.api.java.JavaDStream;
 import org.apache.spark.streaming.api.java.JavaPairDStream;
 import org.apache.spark.streaming.api.java.JavaStreamingContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import java.util.ArrayList;
+import java.util.Collection;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import javax.annotation.Nullable;
 
 /**
  * Driver for running pipelines using Spark Streaming.
  */
 public class SparkStreamingPipelineRunner extends SparkPipelineRunner {
 
+  private static final Logger LOG = LoggerFactory.getLogger(SparkStreamingPipelineRunner.class);
+
   private final JavaSparkExecutionContext sec;
-  private final JavaStreamingContext streamingContext;
+  private final JavaStreamingContext javaStreamingContext;
   private final DataStreamsPipelineSpec spec;
-  private final boolean checkpointsDisabled;
+  private final boolean nativeStateStoreEnabled;
 
-  public SparkStreamingPipelineRunner(JavaSparkExecutionContext sec, JavaStreamingContext streamingContext,
-                                      DataStreamsPipelineSpec spec, boolean checkpointsDisabled) {
+  public SparkStreamingPipelineRunner(JavaSparkExecutionContext sec, JavaStreamingContext javaStreamingContext,
+                                      DataStreamsPipelineSpec spec) {
     this.sec = sec;
-    this.streamingContext = streamingContext;
-    this.checkpointsDisabled = checkpointsDisabled;
+    this.javaStreamingContext = javaStreamingContext;
     this.spec = spec;
+    this.nativeStateStoreEnabled = spec.getStateSpec().getMode() == DataStreamsStateSpec.Mode.NATIVE_STATE_STORE
+      && !spec.isPreviewEnabled(sec);
+    LOG.debug("Native state handling enabled : {}", nativeStateStoreEnabled);
```

Comment:
nit: this is kind of a confusing message when false, let's log enabled/disabled based on the flag instead of logging the true/false value

Reply:
Done